File: solutions/zig/leap/leap.zig
pub fn isLeapYear(year: u32) bool {
    return isDivisibleBy(year, 4) and
        !isDivisibleBy(year, 100) or
        isDivisibleBy(year, 400);
}

fn isDivisibleBy(year: u32, n: u32) bool {
    return year % n == 0;
}

File: solutions/java/leap/src/main/java/Leap.java
interface LambdaLinterface {
    public boolean isDivisibleBy(int n);
}

class Leap {
    public boolean isLeapYear(int year) {
        LambdaLinterface check = (n) -> year % n == 0;
        return check.isDivisibleBy(4) &&
                !check.isDivisibleBy(100) ||
                check.isDivisibleBy(400);
    }
}
